src/space.js

```diff
@@ -53,6 +53,27 @@ export default class ColorSpace {
 			this.formats.color.id = this.id;
 		}
 
+		// Gamut space
+
+		this.gamutSpace = this;
+
+		if (options.gamutSpace) {
+			if (options.gamutSpace !== "self") {
+				this.gamutSpace = ColorSpace.get(options.gamutSpace);
+			}
+		}
+		else if (this.isPolar) {
+			// Do not check gamut through polar coordinates
+			this.gamutSpace = this.base;
+		}
+
+		// Optimize inGamut for unbounded spaces
+		if (this.gamutSpace.isUnbounded){
+			this.inGamut = (coords, options) => {
+				return true;
+			};
+		}
+
```

Comment:
Minor nit: Setting `this.gamutSpace` to `this` and then undoing it later makes the logic hard to follow. I'd restructure a bit to avoid this, possibly via having two branches for the two main cases like:

```js
if (options.gamutSpace) {
	// Gamut space explicitly specified
	...
}
else {
	// Gamut space not explicitly specified, fall back to sensible defaults
	...
}
```

Note that if the gamut space is unbounded, `this.gamutSpace` is effectively still `this`, we just override `inGamut` as a performance optimization.

Reply:
I've changed the code to look more like what you suggested.